### DISCUSSION TITLE
[Support]: Object wasn't detected but a recording of the video was

### DISCUSSION THREAD

**Discussion by Sian-Lee-SA** [blakeblackshear/frigate#11008] [General Support] [answered]
### Describe the problem you are having

While in bed I noticed on my security footage that there was a cat walking around my front door. I checked Frigate to make sure it detected it which it did not. I figured I may have to adjust settings so I downloaded a minute of the clip before the cat enters and after the cat leaves from surveillance station which is the same stream (resolution, fps, codecs etc.). I added the mp4 file to go2rtc then copied the same settings from the front door while only changing the stream pointing to the file stream.

The mp4 detected the cat every time... What's with that?

Resources on my server are plentiful and would not be caused by lack of resources when the event happened live. This is also frustrating because it's not the only time I missed objects when I feel it should've detected them, unfortunately I never followed up with recordings to test on other occasions.

**Go2RTC settings:**
![go2rtc](https://github.com/blakeblackshear/frigate/assets/11685433/9d305914-15da-4162-827b-62154920b21a)

**Front Door Camera Settings:**
![front_door_settings](https://github.com/blakeblackshear/frigate/assets/11685433/008607ce-2449-4697-acd6-7f78662d23b2)

**Test Camera Settings:**
![test_settings](https://github.com/blakeblackshear/frigate/assets/11685433/e04631bf-1603-4faf-b02d-ed895d8fa826)

**Result from test mp4:**
![test_clip_detection](https://github.com/blakeblackshear/frigate/assets/11685433/4c53d54b-588a-46de-a945-bc68b2de43f3)

**Both probes:**
![probe_a](https://github.com/blakeblackshear/frigate/assets/11685433/1f87e06a-674a-43f5-92cb-93f7a6816646)
![probe_b](https://github.com/blakeblackshear/frigate/assets/11685433/036dc0fd-6b27-45cd-80fc-cef166779336)

**Detectors:**
![detectors](https://github.com/blakeblackshear/frigate/assets/11685433/97e66d27-5c81-4233-8d7c-7eca05cf5fc9)

**All Camera Stats:**
![camera_stats](https://github.com/blakeblackshear/frigate/assets/11685433/e5f4bf88-030b-4f1c-b2bb-51bcee15dc05)

**The following is the event history for the camera which you can tell the front door is missing the event but not the tests:**
![event_history](https://github.com/blakeblackshear/frigate/assets/11685433/08aa5004-9d02-40ac-8570-cfd01f007a15)

Would love to get the consistency like the tests so some insight will be appreciated.

### Version

0.13.1-34FB1C2

### Frigate config file

```yaml
Attached picture above
```

### Relevant log output

```shell
Logs have nothing, All [INFO] and TPU started.
```

### FFprobe output from your camera

```shell
Attached picture above
```

### Frigate stats

```json
{"cameras":{"back_yard_outer":{"audio_dBFS":0.0,"audio_rms":0.0,"camera_fps":6.1,"capture_pid":535,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":542,"pid":518,"process_fps":6.1,"skipped_fps":0.0},"front_door":{"audio_dBFS":0.0,"audio_rms":0.0,"camera_fps":10.1,"capture_pid":526,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":531,"pid":514,"process_fps":10.1,"skipped_fps":0.0},"front_yard":{"audio_dBFS":0.0,"audio_rms":0.0,"camera_fps":8.1,"capture_pid":540,"detection_enabled":1,"detection_fps":28.0,"ffmpeg_pid":546,"pid":520,"process_fps":8.1,"skipped_fps":0.0},"open_area":{"audio_dBFS":0.0,"audio_rms":0.0,"camera_fps":9.8,"capture_pid":522,"detection_enabled":1,"detection_fps":1.8,"ffmpeg_pid":527,"pid":513,"process_fps":9.8,"skipped_fps":0.0},"veranda":{"audio_dBFS":0.0,"audio_rms":0.0,"camera_fps":6.1,"capture_pid":530,"detection_enabled":1,"detection_fps":0.0,"ffmpeg_pid":539,"pid":516,"process_fps":6.1,"skipped_fps":0.0}},"cpu_usages":{"1":{"cmdline":"/package/admin/s6/command/s6-svscan -d4 -- /run/service","cpu":"0.0","cpu_average":"0","mem":"0.0"},"15":{"cmdline":"s6-supervise s6-linux-init-shutdownd","cpu":"0.0","cpu_average":"0","mem":"0.0"},"16":{"cmdline":"/package/admin/s6-linux-init/command/s6-linux-init-shutdownd -c /run/s6/basedir -g 3000 -C -B","cpu":"0.0","cpu_average":"0","mem":"0.0"},"24":{"cmdline":"s6-supervise s6rc-oneshot-runner","cpu":"0.0","cpu_average":"0","mem":"0.0"},"25":{"cmdline":"s6-supervise s6rc-fdholder","cpu":"0.0","cpu_average":"0","mem":"0.0"},"26":{"cmdline":"s6-supervise frigate","cpu":"0.0","cpu_average":"0","mem":"0.0"},"27":{"cmdline":"s6-supervise frigate-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"28":{"cmdline":"s6-supervise go2rtc","cpu":"0.0","cpu_average":"0","mem":"0.0"},"29":{"cmdline":"s6-supervise go2rtc-healthcheck","cpu":"0.0","cpu_average":"0","mem":"0.0"},"30":{"cmdline":"s6-supervise go2rtc-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"31":{"cmdline":"s6-supervise nginx","cpu":"0.0","cpu_average":"0","mem":"0.0"},"32":{"cmdline":"s6-supervise nginx-log","cpu":"0.0","cpu_average":"0","mem":"0.0"},"41":{"cmdline":"/package/admin/s6/command/s6-ipcserverd -1 -- /package/admin/s6/command/s6-ipcserver-access -v0 -E -l0 -i data/rules -- /package/admin/s6/command/s6-sudod -t 30000 -- /package/admin/s6-rc/command/s6-rc-oneshot-run -l ../.. --","cpu":"0.0","cpu_average":"0","mem":"0.0"},"42":{"cmdline":"/package/admin/s6-2.11.3.2/command/s6-fdholderd -1 -i data/rules","cpu":"0.0","cpu_average":"0","mem":"0.0"},"80":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/frigate","cpu":"0.0","cpu_average":"0","mem":"0.0"},"81":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/nginx","cpu":"0.0","cpu_average":"0","mem":"0.0"},"82":{"cmdline":"s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/go2rtc","cpu":"0.0","cpu_average":"0","mem":"0.0"},"89":{"cmdline":"/usr/local/go2rtc/bin/go2rtc -config=/dev/shm/go2rtc.yaml","cpu":"4.0","cpu_average":"3","mem":"0.1"},"101":{"cmdline":"bash ./run.user go2rtc-healthcheck","cpu":"0.0","cpu_average":"0","mem":"0.0"},"102":{"cmdline":"python3 -u -m frigate","cpu":"1.9","cpu_average":"1","mem":"1.6"},"119":{"cmdline":"nginx: master process nginx","cpu":"0.0","cpu_average":"0","mem":"0.1"},"145":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"146":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"147":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"148":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"149":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"150":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"156":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"179":{"cmdline":"nginx: worker process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"204":{"cmdline":"nginx: cache manager process","cpu":"0.0","cpu_average":"0","mem":"0.0"},"439":{"cmdline":"frigate.logger       ","cpu":"0.0","cpu_average":"0","mem":"0.6"},"478":{"cmdline":"frigate.recording_manager","cpu":"0.1","cpu_average":"0","mem":"0.8"},"487":{"cmdline":"/usr/bin/python3 -c from multiprocessing.resource_tracker import main;main(57)","cpu":"0.4","cpu_average":"0","mem":"0.1"},"488":{"cmdline":"frigate.detector.coral","cpu":"6.9","cpu_average":"5","mem":"0.9"},"490":{"cmdline":"frigate.output       ","cpu":"1.5","cpu_average":"1","mem":"0.8"},"496":{"cmdline":"ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1280x720 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.1"},"497":{"cmdline":"ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1280x720 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.1"},"498":{"cmdline":"ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1920x1080 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.1"},"499":{"cmdline":"ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1920x1080 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.1"},"500":{"cmdline":"ffmpeg -f rawvideo -pix_fmt yuv420p -video_size 1280x720 -i pipe: -f mpegts -s 1280x720 -codec:v mpeg1video -q 8 -bf 0 pipe:","cpu":"0.0","cpu_average":"0","mem":"0.1"},"513":{"cmdline":"frigate.process:open_area","cpu":"2.1","cpu_average":"3","mem":"0.9"},"514":{"cmdline":"frigate.process:front_door","cpu":"1.5","cpu_average":"1","mem":"0.9"},"516":{"cmdline":"frigate.process:veranda","cpu":"0.8","cpu_average":"1","mem":"0.9"},"518":{"cmdline":"frigate.process:back_yard_outer","cpu":"1.0","cpu_average":"1","mem":"1.1"},"520":{"cmdline":"frigate.process:front_yard","cpu":"11.6","cpu_average":"6","mem":"1.0"},"522":{"cmdline":"frigate.capture:open_area","cpu":"1.8","cpu_average":"1","mem":"0.8"},"526":{"cmdline":"frigate.capture:front_door","cpu":"2.0","cpu_average":"1","mem":"0.8"},"527":{"cmdline":"ffmpeg -hide_banner -loglevel warning -threads 2 -user_agent FFmpeg Frigate/0.13.1-34fb1c2 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://127.0.0.1:8554/camera.rtsp_open_area -r 10 -vf fps=10,scale=1280:720 -threads 2 -f rawvideo -pix_fmt yuv420p pipe:","cpu":"11.8","cpu_average":"12","mem":"0.3"},"530":{"cmdline":"frigate.capture:veranda","cpu":"2.2","cpu_average":"2","mem":"0.8"},"531":{"cmdline":"ffmpeg -hide_banner -loglevel warning -threads 2 -user_agent FFmpeg Frigate/0.13.1-34fb1c2 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://127.0.0.1:8554/camera.rtsp_front_door -r 10 -vf fps=10,scale=1280:720 -threads 2 -f rawvideo -pix_fmt yuv420p pipe:","cpu":"3.8","cpu_average":"3","mem":"0.2"},"535":{"cmdline":"frigate.capture:back_yard_outer","cpu":"2.2","cpu_average":"2","mem":"0.8"},"539":{"cmdline":"ffmpeg -hide_banner -loglevel warning -threads 2 -user_agent FFmpeg Frigate/0.13.1-34fb1c2 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://127.0.0.1:8554/camera.rtsp_veranda -r 6 -vf fps=6,scale=1920:1080 -threads 2 -f rawvideo -pix_fmt yuv420p pipe:","cpu":"9.9","cpu_average":"10","mem":"0.3"},"540":{"cmdline":"frigate.capture:front_yard","cpu":"1.4","cpu_average":"1","mem":"0.8"},"542":{"cmdline":"ffmpeg -hide_banner -loglevel warning -threads 2 -user_agent FFmpeg Frigate/0.13.1-34fb1c2 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://127.0.0.1:8554/camera.rtsp_back_yard_outer -r 6 -vf fps=6,scale=1920:1080 -threads 2 -f rawvideo -pix_fmt yuv420p pipe:","cpu":"11.8","cpu_average":"12","mem":"0.3"},"546":{"cmdline":"ffmpeg -hide_banner -loglevel warning -threads 2 -user_agent FFmpeg Frigate/0.13.1-34fb1c2 -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -timeout 5000000 -use_wallclock_as_timestamps 1 -i rtsp://127.0.0.1:8554/camera.rtsp_front_yard -r 8 -vf fps=8,scale=1280:720 -threads 2 -f rawvideo -pix_fmt yuv420p pipe:","cpu":"5.6","cpu_average":"5","mem":"0.2"},"2007":{"cmdline":"sleep 30s","cpu":"0.0","cpu_average":"0","mem":"0.0"}},"detection_fps":29.8,"detectors":{"coral":{"detection_start":1713376542.050622,"inference_speed":7.59,"pid":488}},"processes":{"go2rtc":{"pid":89},"logger":{"pid":439},"recording":{"pid":478}},"service":{"last_updated":1713376542,"latest_version":"0.13.2","storage":{"/dev/shm":{"free":241.8,"mount_type":"tmpfs","total":256.0,"used":14.2},"/media/frigate/clips":{"free":141526.8,"mount_type":"btrfs","total":224415.5,"used":82888.6},"/media/frigate/recordings":{"free":141526.8,"mount_type":"btrfs","total":224415.5,"used":82888.6},"/tmp/cache":{"free":953.7,"mount_type":"tmpfs","total":953.7,"used":0.0}},"temperatures":{},"uptime":1930,"version":"0.13.1-34fb1c2"}}
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Dahua DH-IPC-HDW3649H-AS-PV-ANZ

**Comment by NickM-27** [comment]
The results are not deterministic which means the results (motion boxes, object boxes) won't be the same every time. One thing to consider here is the length of the test clip. How much time are you giving it before the cat is in view? 

One possible explanation is that the average motion frame that frigate had is different than what happens when the short clip is ran so the cat was detected then. 

Another possibility is that your region grid is too large because it only saw people so the region chosen was too large. The region grid doesn't apply to your `test` camera because the new name test means it does not have this region grid. You can test this by opening `frigate:5000/api/front_door/grid.jpg`

**Comment by Sian-Lee-SA** [comment]
The test clip was about 20 seconds before cat enters and 20 seconds after (didn't really want to sit around and wait if it was a matter of tweeking some settings). It's something I need to debug live, unfortunately we don;t get that luxury with animals and I was relying on animal detection for deterrence by turning on the irrigation for a few seconds when they passed a certain zone (not applicable in this scenario).

I took a screenshot of the grid (looks like chinese matrix to me) is the following image a cause for issue?
![Screenshot 2024-04-18 035227](https://github.com/blakeblackshear/frigate/assets/11685433/2e6ef092-247b-41c7-a9bb-0eb55d99395f)

**Reply by NickM-27** [reply]
what model are you running?

**Reply by Sian-Lee-SA** [reply]
default
only added the following
```yaml
model:
  labelmap:
    17: cat
```
* Sorry, replied in wrong box earlier

**Reply by NickM-27** [reply]
okay, so if we look at the area where the cat is in your screenshot we can see that the `avg` region size is 595 pixels with a standard deviation of 120 pixels. This means that most likely when the cat motion was detected the region was made larger because typically this camera sees people in that portion of the image. 

You should be able to confirm this by playing the test clip on the same front door camera. The regions (green boxes on the debug view) will be larger than on the test cam.

For starters you could try resetting the regions DB, I can help instruct how to do that if needed. Otherwise you can show some examples of the cat clip to that specific camera and it will get more examples of the cat and adjust the grid to better account for objects of that size

**Reply by Sian-Lee-SA** [reply]
Yep, you're spot on about that. Region boxes are miles apart lol...
![image](https://github.com/blakeblackshear/frigate/assets/11685433/e0355ebd-6141-40fc-8bc7-575992dc0eec)
![image](https://github.com/blakeblackshear/frigate/assets/11685433/2ec841c5-fec3-49b9-b030-c1067c654ebc)

Would the the second option be more viable? Do I Just keep playing the video over and over (kinda like training)? Or would a reset just be better overall?

**Reply by NickM-27** [reply, marked as answer]
the problem with a reset is it will reload based on the events currently in your database. Those will most likely just be people since you said animals appear very seldom so you would most likely benefit from playing the video some just so it can adjust the sizing. 

I also want to make clear that this isn't really intended obviously. On a more typical "outwards" camera perspective the region DB works well to anticipate the sizes of objects that are detected and reduce false positives due to a more accurate size region being used. This downwards perspective does seem to be an exception to that where the middle of an object right up close to the camera will be the same as the middle of a smaller object like you are showing here.

**Reply by Sian-Lee-SA** [reply]
I'll do playing the video a go, everything you said makes sense and is understandable as people entering the door takes up half the image while detected. I had the same thought that the rarity of animals (cats or even dogs) coming by would just cause it to default back to how it is. I'm well versed with SQL (that's if the DB, is SQLite) so I can tackle the DB if need be. Will see how this goes with the clip(s) (will save more down the line when they happen).

Cheers

**Reply by NickM-27** [reply]
right, it is sqlite, the table name is REGIONS and the there is a camera field so you can just delete the entry for that camera. But worth trying what you are doing first. I'll be curious to know if it the looping does cause it to be detected.

**Reply by Sian-Lee-SA** [reply]
Will let you know when I've don't feel like a zombie. 4AM here now and tired asf. Can't even spell properly

**Reply by Sian-Lee-SA** [reply]
Left it running for the rest of the night and would detect it every 30 minutes or so, figured I would just delete the entry in the DB. Used SQLite Studio but for some reason it just corrupted the DB after making the commit `peewee.DatabaseError: database disk image is malformed`. Will restore from a backup and if I still fail on that I'll just nuke the file, Frigate is only used for notifying HA so losing the event records etc isn't a problem.

Edit: I think because the file was being used by Frigate while I was viewing the DB was the cause even though I shut down Frigate and did a reconnection in SQLiteStudio then deleted the entry. Either way, restored a backup and tried again and was fine. See how it goes from here. Might script something in future to help mitigate the issue.
